cmd/run.go

```diff
@@ -95,15 +96,50 @@ func bindRunFlags(cmd *cobra.Command, config *app.Config) {
 	}
 }
 
+func bindPrivKeyFlag(cmd *cobra.Command, privKeyFile *string) {
+	var (
+		emptyStr = ""
+		dataDir  = &emptyStr
+	)
+
+	cmd.Flags().StringVar(dataDir, "data-dir", "", "The directory where charon stores all its internal data. Deprecated.")
+	cmd.Flags().StringVar(privKeyFile, "private-key", ".charon/charon-enr-private-key", "The path to the charon enr private key.")
+
+	preRunE := cmd.PreRunE // Allow multiple wraps of PreRunE.
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		ctx := log.WithTopic(cmd.Context(), "cmd")
+		if *dataDir != "" {
+			log.Warn(ctx, "Deprecated flag 'data-dir' used, please use new flag 'private-key'.", nil)
+		}
+
+		if _, err := os.Open(*privKeyFile); err == nil {
+			return nil
+		}
+
+		if *dataDir == "" {
+			return errors.New("private key file not found")
+		}
+
+		keyPath := p2p.KeyPath(*dataDir)
+		if _, err := os.Open(keyPath); errors.Is(err, os.ErrNotExist) {
+			return errors.New("private key file not found")
+		}
+
+		*privKeyFile = keyPath
+
+		if preRunE != nil {
+			return preRunE(cmd, args)
+		}
```

Comment:
if you return early, you are not calling preRunE... so multiple wraps doesn't work...
```
if _, err := os.Open(*privKeyFile); err == nil {
  // Ignore data-dir since priv key file present
} else if  _, err := os.Open(p2p.KeyPath(dataDir)); err == nil {
  // Use deprecated data-dir
  *privKeyFile = p2p.KeyPath(dataDir)
} else {
  return errors.New("charon enr private key file not found in either `data-dir` or `private-key-file`")
}

if preRunE != nil {
 return preRunE(cmd, args)
}

return nil
```